app/api/migrations/migrations/49-add_system_key_translations/index.js

```diff
@@ -0,0 +1,65 @@
+import * as fs from 'fs';
+
+import csv from 'api/csv/csv';
+
+/*
+This migration is meant to be repeatable.
+After copy pasting:
+  - change the contents of system_keys.csv to the new keyset
+  - change the file location in the readCsvToSystemKeys call
+  - change the tests, if necessary
+*/
+
+// eslint-disable-next-line max-statements
+async function readCsvToSystemKeys(db, filename) {
+  const fstream = fs.createReadStream(filename);
+  const rows = await csv(fstream).read();
+  fstream.close();
+
+  const translations = await db
+    .collection('translations')
+    .find()
+    .toArray();
+  const locales = translations.map(tr => tr.locale);
+
+  const locToSystemContext = {};
+  translations.forEach(tr => {
+    locToSystemContext[tr.locale] = tr.contexts.find(c => c.id === 'System');
+  });
+  const locToKeys = {};
+  Object.entries(locToSystemContext).forEach(([loc, context]) => {
+    locToKeys[loc] = new Set(context.values.map(v => v.key));
```

Comment:
I usually go for hashed data structures, when I know I need to check that a value exist in the set. In this case we could probably get away with the worse running times, if we think that the input (rows, number of keys) is relatively small. But I think it's a good habit to do it like this even if we think that the optimization is not important, since the code is (in my opinion) not more complicated, and we can be sure that this will perform well on large inputs.

Reply:
It's not important at all, It's fine if you leave that as is.